### PR TITLE
Added Timer and Stall opcodes

### PR DESCRIPTION
**Pull request by Itay2805**

### CODE REVIEW

File: include/lai/host.h [hunk 1]
```diff
@@ -57,7 +57,9 @@ __attribute__((weak)) uint16_t laihost_pci_readw(uint16_t, uint8_t, uint8_t, uin
 __attribute__((weak)) void laihost_pci_writed(uint16_t, uint8_t, uint8_t, uint8_t, uint16_t, uint32_t);
 __attribute__((weak)) uint32_t laihost_pci_readd(uint16_t, uint8_t, uint8_t, uint8_t, uint16_t);
 
+__attribute__((weak)) void laihost_stall(uint64_t);
 __attribute__((weak)) void laihost_sleep(uint64_t);
+__attribute__((weak)) uint64_t laihost_timer(void);
```
Comment (avdgrinten) [inline include/lai/host.h:62]: Stall() and Timer() should use the same `laihost` function -- there is no need to make this customizable.

File: core/exec.c [hunk 2]
```diff
@@ -1070,6 +1070,30 @@ static lai_api_error_t lai_exec_reduce_op(int opcode, lai_state_t *state, struct
 
         break;
     }
+    case (EXTOP_PREFIX << 8) | STALL_OP: {
+        if (!laihost_stall && !laihost_timer)
+            lai_panic("host does not provide timer functions required by Stall()");
+
+        lai_variable_t time = {0};
```
Comment (Sebastian-byte) [inline core/exec.c:1077]: Shouldn't LAI variables be declared as `LAI_CLEANUP_VAR lai_variable_t name = LAI_VAR_INITIALIZER`?

Comment (Itay2805) [reply core/exec.c:1077]: I think it is not needed for integers

File: include/lai/host.h (hunk 1 above)
Comment (Itay2805) [reply include/lai/host.h:62]: Ok, will do it tonight

File: core/exec.c (hunk 2 above)
Comment (thomtl) [reply core/exec.c:1077]: It's not needed for integers per se, but it should be done like that, to leave space to change that behaviour in the future

Comment (Itay2805) [reply core/exec.c:1077]: Well, iirc this was copied from sleep, should I add it there as well?

**Comment by Sebastian-byte** [comment]
I suppose this can be merged now?

**Review by avdgrinten — APPROVED** [review]
LGTM.

**Pull request merged**